src/app/(with-sidebar)/myProfile/page.tsx

```diff
@@ -101,19 +123,50 @@ const MyProfile = () => {
     handleLikeButton,
     addComment,
     hasMorePosts,
+    userLikedPosts,
   ]);
   return (
     <Grid columns={1}>
       <ProfileCard
         coverImage="https://picsum.photos/200"
-        bio="bio"
-        followers="1M"
-        posts={postDataWithComments.length}
+        bio={userInfo.bio ? userInfo.bio : ""}
+        followers={followerCount}
+        posts={postCount}
         profileImage={userProfile}
-        name="kalamanch"
+        name={userInfo.name}
+        handleEditProfileOpen={handleEditProfileOpen}
       />
       <CustomTabs tabs={tabs} activeTab={tab} onTabChange={handleChange} />
       {renderUI}
+      {isEditProfileOpen && (
+        <EditProfile
+          open={isEditProfileOpen}
+          handleClose={handleEditProfileClose}
+          profileData={userInfo}
+          handleProfileSave={async ({
+            name,
```

Comment:
use a type here and move all this logic to hook, just expose a callback

Reply:
done